RapidWebView/src/main/java/in/workindia/rapidwebview/utils/RapidBrodCastActionUtility.kt

```diff
@@ -0,0 +1,183 @@
+                generateJavaScriptEvent(
+                    "rapid-web-view-upload-listener", mapOf(
+                        "status" to "failure"
+                    )
+                )
+            }
+
+            permissionStatus.equals(BroadcastConstants.SUCCESS, ignoreCase = true) -> {
+                permissionSuccessHelper(permissionList, intent, context)
+            }
+
+            permissionStatus.equals(BroadcastConstants.FAILURE, ignoreCase = true) -> {
+                generateJavaScriptEvent(
+                    "rapid-web-view-permission-listener", mapOf(
+                        "status" to "failure"
+                    )
+                )
+            }
+
+            else -> ""
+        }
+
+        if (javaScript.isNotBlank()) {
+            dispatchJavaScript(javaScript, webView)
+        }
+    }
+
+
+    /**
+     * Handles the action to be taken when a download is complete.
+     * Displays a toast message and opens the downloaded file.
+     *
+     * @param context The context received from BrodCastReceiver.
+     * @param intent The intent received from BrodCastReceiver.
+     */
+    fun handleDownloadCompleteAction(context: Context?, intent: Intent) {
+        Toast.makeText(context, "Download Complete", Toast.LENGTH_SHORT).show()
+        val downloadId = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1)
+        context?.let {
+            RapidDownLoadUtility.openDownLoadedFile(it, downloadId)
+        }
+    }
+
+
+    /**
+     * Helper function to handle successful permission grants. It checks if
+     * the required permissions for reading and writing files are granted,
+     * and if so, proceeds with checking the [DOWNLOAD_URL] from intent and initiates a file download.
+     *
+     * we ask permissions when [RapidWebViewJSInterface.downLoadFileLocally] and [RapidWebViewJSInterface.downLoadFileLocally] are called. we continue the download only if the permissions are granted.
+     * @see [RapidWebViewJSInterface.downLoadFileLocally]
+     * @see [RapidWebViewJSInterface.downLoadAndOpenFile]
+     *
+     * @param permissionList List of granted permissions.
+     * @param intent The intent received from BrodCastReceiver.
+     * @param context The context received from BrodCastReceiver.
+     * @return A JavaScript event string indicating the success of the permission grant.
+     */
+    private fun permissionSuccessHelper(
+        permissionList: Array<String>,
+        intent: Intent,
+        context: Context?,
+    ): String {
+        when {
+            permissionList.contentEquals(RapidPermissionHelper.requiredPermissionsForReadAndWrite()) -> {
+                checkAndDownloadFile(intent, context)
+            }
+        }
+
+        return generateJavaScriptEvent(
+            "rapid-web-view-permission-listener", mapOf(
+                "status" to "success",
+                "permissionList" to permissionList.joinToString(", ")
+            )
+        )
```

Comment:
This function is no longer following single responsibility principle. Rework is required.